internal/controllers/machineset/machineset_controller.go

```diff
@@ -306,16 +306,13 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 			continue
 		}
 		if conditions.IsFalse(machine, clusterv1.MachineOwnerRemediatedCondition) {
-			log.Info("Deleting machine because marked as unhealthy by the MachineHealthCheck controller")
-			patch := client.MergeFrom(machine.DeepCopy())
-			if err := r.Client.Delete(ctx, machine); err != nil {
-				errs = append(errs, errors.Wrap(err, "failed to delete"))
-				continue
-			}
-			conditions.MarkTrue(machine, clusterv1.MachineOwnerRemediatedCondition)
-			if err := r.Client.Status().Patch(ctx, machine, patch); err != nil && !apierrors.IsNotFound(err) {
-				errs = append(errs, errors.Wrap(err, "failed to update status"))
-			}
+			log.Info("Deleting machine because marked as unhealthy by the MachineHealthCheck controller: %v", machine.Name)
+			errs = append(r.deleteMachine(ctx, machine))
+		}
+		// The delete-machines annotation may be set by external sources, such as Cluster Autoscaler
+		if _, ok := machine.ObjectMeta.Annotations[clusterv1.DeleteMachineAnnotation]; ok {
```

Comment:
I think if this route is chosen there should be a new annotation created specifically for this purpose.. 

`clusterv1.DeleteMachineAnnotation` is already used below in syncReplicas in L567 to give priority to deletion. Using the same annotation here changes the behaviour in a way that's confusing.

Reply:
Is the delete-machine annotation set by anything other than CA?

Reply:
I don't know -  but it's a public annotation with a defined behaviour so I don't think we should change it here.

Reply:
Sounds reasonable. I've added a new annotation `cluster.x-k8s.io/machine-marked-for-delete`.